botocore/loaders.py

```diff
@@ -353,7 +389,7 @@ def load_service_model(self, service_name, type_name, api_version=None):
         return self.load_data(full_path)
 
     @instance_cache
-    def load_data(self, name):
+    def load_data(self, name, use_ordered_dict=True):
```

Comment:
3

My reasons being:

1. This will apply to not just endpoints loading.  In fact, the only thing we actually use ordered dict for (that I can remember) is for consistent ordering in the docs for member names.  I'm not even sure botocore needs ordered dicts internally.  We might be able to default to dict() and have doc generation figure out some way to configure ordered dicts for html generation (not sure about inline docstrings in a repl)

2. We haven't really figured out how we're going to do text/rst file loading, and how that's implemented could change how (2) would be implemented.